enarx-keep-sev/src/vm/mem.rs

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pub use kvm_bindings::kvm_userspace_memory_region as KvmUserMemoryRegion;
+use mmap::Unmap;
+use span::{Contains, Span};
+use x86_64::{PhysAddr, VirtAddr};
+
+use std::io;
+
+pub struct Region {
+    _kvm_region: KvmUserMemoryRegion,
+    _unmap: Unmap,
+}
+
+impl Region {
+    pub fn new(kvm_region: KvmUserMemoryRegion, unmap: Unmap) -> Self {
+        Self {
+            _kvm_region: kvm_region,
+            _unmap: unmap,
+        }
+    }
+}
+
+pub type AddressSpace = Vec<Region>;
```

Comment:
I have a sneaking suspicion that this type really belongs as a sub-feature of a `VirtualMachine` type instead.